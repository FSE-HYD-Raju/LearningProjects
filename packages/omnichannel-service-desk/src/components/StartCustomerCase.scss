.StartCustomerCase {
	display: flex;
	justify-content: center;
}
