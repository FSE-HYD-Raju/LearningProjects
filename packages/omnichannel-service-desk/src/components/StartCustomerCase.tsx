import { cssns, FormattedMessage, OcButton, OcButtonSize, OcButtonType } from "omnichannel-common-pos";
import { withRouter, RouteComponentProps } from "react-router";
import { Component } from "react";
import messages from "./ServiceDesk.messages";

const { React } = cssns("StartCustomerCase");

interface StartCustomerCaseStateProps {
	actions: {
		createNewCustomerCase: (customerId: string) => void;
	};
	salesRepId: string;
	toolmode: boolean;
}

interface StartCustomerCaseState {
	createCaseBtnDisabled: boolean;
}

type StartCustomerCaseProps = StartCustomerCaseStateProps & RouteComponentProps<any>;

class StartCustomerCase extends Component<StartCustomerCaseProps, StartCustomerCaseState> {

	constructor(props: StartCustomerCaseProps) {
		super(props);
		this.state = {
			createCaseBtnDisabled: false
		};
	}

	startNewCustomerCase = () => {
		this.setState({
			createCaseBtnDisabled: true
		});
		this.props.actions.createNewCustomerCase(this.props.salesRepId);
		this.props.history.push("/servicedesk/customer");
	};

	render() {
		return (
			<div className="this w-box">
				<OcButton
					buttonType={OcButtonType.SUCCESS}
					buttonSize={OcButtonSize.LARGE}
					id="buttonStartNewCustomerCase"
					onClick={this.startNewCustomerCase}
					disabled={this.state.createCaseBtnDisabled || this.props.toolmode}
				>
					<FormattedMessage {...messages.startNewCustomerCase}/>
				</OcButton>
			</div>
		);
	}
}

export default withRouter(StartCustomerCase);
export {
	StartCustomerCaseStateProps,
	StartCustomerCaseProps,
	StartCustomerCaseState,
	StartCustomerCase as StartCustomerCaseRaw,
};
