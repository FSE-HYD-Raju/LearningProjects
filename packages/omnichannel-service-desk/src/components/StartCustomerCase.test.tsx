import * as React from "react";

import {
	mockRouterProps,
	mountWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps
} from "omnichannel-common-pos";

import { StartCustomerCaseRaw } from "./StartCustomerCase";

describe("StartCustomerCase", () => {
	const minProps = {
		actions: {
			createNewCustomerCase: jest.fn(),
		},
		toolmode: false,
		salesRepId: "sales-rep-id",
		location: mockRouterProps.location,
		history: mockRouterProps.history,
		match: mockRouterProps.match,
		staticContext: mockRouterProps.staticContext,
	};

	it("succeeds at shallow mount with min props", () => {
		const wrapper = shallowWithContext(<StartCustomerCaseRaw {...minProps} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount with min props", () => {
		mountWithContext(<StartCustomerCaseRaw {...minProps} />);
	});

	it("renders with minimum props", () => {
		const wrapper = mountWithContext(
			<StartCustomerCaseRaw
				{...minProps}
			/>
		);

		const button = wrapper.find("#buttonStartNewCustomerCase").hostNodes();
		expect(button).toHaveLength(1);
		expect(wrapper.find("#buttonStartNewCustomerCase").hostNodes()).toHaveLength(1);
		expect(button.text().toLowerCase()).toEqual("Start new customer case".toLowerCase());
	});

	it("calls given createNewCustomerCase() with the id of current sales representative", () => {
		const salesRepId = "666";

		const props = {
			...minProps,
			salesRepId,
			actions: {
				createNewCustomerCase: jest.fn(),
			}
		};

		const createNewCustomerCaseSpy = props.actions.createNewCustomerCase;

		const wrapper = mountWithContextAndRouterProps(
			<StartCustomerCaseRaw {...props} />
		);

		const button = wrapper.find("#buttonStartNewCustomerCase").hostNodes();
		button.simulate("click");
		expect(createNewCustomerCaseSpy).toHaveBeenCalledWith(salesRepId);
	});
});
