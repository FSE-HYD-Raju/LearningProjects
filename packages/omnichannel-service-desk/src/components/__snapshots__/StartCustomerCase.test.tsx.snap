// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartCustomerCase succeeds at shallow mount with min props 1`] = `
<div
  className="StartCustomerCase w-box"
>
  <OcButton
    buttonSize="LARGE"
    buttonType="SUCCESS"
    disabled={false}
    id="buttonStartNewCustomerCase"
    onClick={[Function]}
  >
    <FormattedMessageWrapper
      defaultMessage="Start new customer case"
      description="Used with button that start's new customer case on POS"
      id="start-new-customer-case"
      wrapperType="wrapper"
    />
  </OcButton>
</div>
`;
